sdk/python/lib/pulumi/resource.py

```diff
@@ -858,24 +875,79 @@ def __init__(self,
                  pkg: str,
                  name: str,
                  props: Optional[dict] = None,
-                 opts: Optional[ResourceOptions] = None) -> None:
+                 opts: Optional[ResourceOptions] = None,
+                 dependency: bool = False) -> None:
         """
         :param str pkg: The package type of this provider resource.
         :param str name: The name of this resource.
         :param Optional[dict] props: An optional list of input properties to use as inputs for the resource.
         :param Optional[ResourceOptions] opts: Optional set of :class:`pulumi.ResourceOptions` to use for this
                resource.
+        :param bool dependency: True if this is a synthetic resource used internally for dependency tracking.
         """
 
         if opts is not None and opts.provider is not None:
             raise TypeError(
                 "Explicit providers may not be used with provider resources")
         # Provider resources are given a well-known type, prefixed with "pulumi:providers".
         CustomResource.__init__(
-            self, f"pulumi:providers:{pkg}", name, props, opts)
+            self, f"pulumi:providers:{pkg}", name, props, opts, dependency)
         self.package = pkg
 
 
+class DependencyResource(CustomResource):
+    """
+    A DependencyResource is a resource that is used to indicate that an Output has a dependency on a particular
+    resource. These resources are only created when dealing with remote component resources.
+    """
+
+    def __init__(self, urn: str) -> None:
+        super().__init__("", "", {}, None, True)
```

Comment:
This line is pretty cryptic with positional params. Can you use named params instead?